tests/unit/post_start_script_test.yaml

```diff
@@ -0,0 +1,285 @@
+          user: yelnats # stanley backwards for the test
+          # ssh_key_file should be /home/yelnats/.ssh/stanley_rsa
+    asserts:
+      - hasDocuments:
+          count: 2
+      - isKind:
+          of: ConfigMap
+      - isAPIVersion:
+          of: v1
+      - isNotEmpty:
+          path: data.[post-start.sh]
+      - equal:
+          path: data.[post-start.sh]
+          value: |
+            #!/bin/bash
+            mkdir -p /home/yelnats/.ssh
+            cp -L /home/yelnats/.ssh-key-vol/stanley_rsa /home/yelnats/.ssh/stanley_rsa
+            chown -R yelnats:yelnats /home/yelnats/.ssh
+            chmod 400 /home/yelnats/.ssh/stanley_rsa
+            chmod 500 /home/yelnats/.ssh
+
+  - it: custom post-start script
+    template: configmaps_post-start-script.yaml
+    set:
+      st2:
+        packs: { sensors: [] } # ensure only 1 sensor
+      st2web:
+        postStartScript: &custom_post_start_script echo hello world
+      st2auth:
+        postStartScript: *custom_post_start_script
+      st2api:
+        postStartScript: *custom_post_start_script
+      st2stream:
+        postStartScript: *custom_post_start_script
+      st2rulesengine:
+        postStartScript: *custom_post_start_script
+      st2timersengine:
+        postStartScript: *custom_post_start_script
+      st2workflowengine:
+        postStartScript: *custom_post_start_script
+      st2scheduler:
+        postStartScript: *custom_post_start_script
+      st2notifier:
+        postStartScript: *custom_post_start_script
+      st2actionrunner:
+        postStartScript: *custom_post_start_script
+      st2sensorcontainer:
+        postStartScript: *custom_post_start_script
+      st2garbagecollector:
+        postStartScript: *custom_post_start_script
+      st2client:
+        postStartScript: *custom_post_start_script
+      st2chatops:
+        enabled: true
+        postStartScript: *custom_post_start_script
+    asserts:
+      - hasDocuments:
+          count: 14
+      - isKind:
+          of: ConfigMap
+      - isAPIVersion:
+          of: v1
+      - isNotEmpty:
+          path: data.[post-start.sh]
+      - matchRegex:
+          path: data.[post-start.sh]
+          # (?m) = multi-line mode: ^ and $ match begin/end line in addition to begin/end text
+          # (?s) = let . match \n
+          # .*? = any character zero or more times, prefer fewer
+          pattern: '(?ms)^#!/bin/bash$.*?^echo hello world$'
+
+  - it: Deployments st2actionrunner and st2client have lifecycle.postStartScipt by default
+    template: deployments.yaml
+    set:
+      st2:
+        packs: { sensors: [] } # ensure only 1 sensor
+      st2chatops:
+        enabled: true
+    asserts:
+      - hasDocuments:
+          count: 14
+
+      # st2actionrunner and st2client do not have checksum annotations
+      # (even though they probably should)
+      - isNull: &assert_checksum
+          path: spec.template.metadata.annotations.[checksum/post-start-script]
```

Comment:
Fix this in a follow up (need to drop the condition since they always have a script)